### PR TITLE
Fix gte lte

### PR DESCRIPTION
**Pull request by sdboyer**
re: #16

### CODE REVIEW

File: constraints_test.go [hunk 1]
```diff
@@ -166,11 +166,28 @@ func TestConstraintCheck(t *testing.T) {
 		{"<=1.1", "0.1.0", true},
 		{"<=1.1", "1.1.0", true},
 		{"<=1.1", "1.1.1", false},
+		{"<=1.1-alpha1", "1.1", false},
+		{"<=2.x", "3.0.0", false},
+		{"<=2.x", "2.9.9", true},
+		{"<2.x", "2.0.0", false},
```
Comment (mattfarina) [inline constraints_test.go]: Why is `<2.x` false with `2.0.0` if `2.x` expands to `<3.0.0`? Am I missing something?

File: constraints_test.go [hunk 2]
```diff
@@ -166,11 +166,28 @@ func TestConstraintCheck(t *testing.T) {
...
+		{"<2.x", "1.9.9", true},
+		{">=2.x", "3.0.0", true},
+		{">=2.x", "2.9.9", true},
+		{">=2.x", "1.9.9", false},
+		{">2.x", "3.0.0", true},
+		{">2.x", "2.9.9", false},
```
Comment (mattfarina) [inline constraints_test.go]: In the `>2.x` case shouldn't it be evaluated as `>2.0.0`? In that case wouldn't `2.9.9` return `true`?

File: constraints_test.go (hunk 1 above)
Comment (sdboyer) [inline constraints_test.go]: Maybe, or maybe I'm missing something. The semantics I'm operating on interpret `x` as "any nonnegative integer". So I take `<2.x` to mean "less than ANY possible valid value for `2.x`". `2.0.0` is a valid possible value for `2.x`, so it should fail.

No? I haven't actually looked at other semver libs to see how they deal with this, or if they even allow gt/lt with wildcard operators at all.

File: constraints_test.go (hunk 2 above)
Comment (sdboyer) [inline constraints_test.go]: Pretty much the same explanation as above - I take `x` to mean "any nonnegative integer", `2.x` to be equivalent to `2.x.x`, and that includes `2.9.9`, so it fails.

File: constraints_test.go (hunk 1 above)
Comment (mattfarina) [inline constraints_test.go]: Oh, I see. Somehow I mentally mushed `<` and `<=`. My bad.

File: constraints_test.go (hunk 2 above)
Comment (mattfarina) [inline constraints_test.go]: Same as other comment.

File: constraints_test.go (hunk 1 above)
Comment (sdboyer) [inline constraints_test.go]: yeah, i think that's what i did when i first wrote this, too. it was a PITA writing this patch, too, i had to whiteboard it out to keep them from mushing in my head 😄

**Pull request merged**